CDCgov/prime-reportstream
# Fixed Simple Report sender transform and timezone offsets

This PR fixes a bug where specimen received time wasn't being set correctly. I The current transform is subtracting an additional 15 minutes from DiagnosticReport.effective and mapping that to Specimen received time which should only add 6 seconds.
It also fixes a bug where the timezone offset was not being set correctly to use daylight savings offset.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Check out and build the branch
2. run `./gradlew reloadSettings`
3. Send a test report using the attached Postman collection
[Simple Report Bug Fix.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/11870198/Simple.Report.Bug.Fix.postman_collection.txt)
4. Verify that the request was accepted
5. Open the generated HL7 file in the local sftp folder.
6. Verify that dates have the PDT -0700 timezone offset.
7. Verify the date in  SPM-18 equals SPM-17 + 6 seconds

## Changes
- Fixes a bug on Simple Report sender transform
- Fixes when calculating timezone offset 

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #10111 

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
